src/t8_forest/t8_forest.cxx

```diff
@@ -2035,13 +2035,31 @@ t8_forest_tree_is_local (const t8_forest_t forest, const t8_locidx_t local_tree)
 int
 t8_forest_element_is_leaf (const t8_forest_t forest, const t8_element_t *element, const t8_locidx_t local_tree)
 {
-  T8_ASSERT (t8_forest_is_committed (forest));
+  bool check_ghost = false;
   T8_ASSERT (t8_forest_tree_is_local (forest, local_tree));
+  return t8_forest_element_is_leaf_or_ghost (forest, element, local_tree, check_ghost);
+}
+
+int
+t8_forest_element_is_leaf_or_ghost (const t8_forest_t forest, const t8_element_t *element, const t8_locidx_t local_tree,
+                                    const int check_ghost)
```

Comment:
I like your idea of seperating the functionality of choosing the element array and searching the element array.

But I think this approach can be done even stricter, by passing the element_array as argument to the helper function.

`t8_forest_element_is_leaf` and `t8_forest_element_is_ghost` then have the task to chose the correct element array, which they then pass to the helper function, which determines if the element is included in the element array.